src/robot.coffee

```diff
@@ -247,6 +264,24 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    @middleware.receive.execute(
+      {response: new Response(this, message)}
+      @processListeners.bind this
+      cb
+    )
+
+  # Private: Passes the given message to any interested Listeners.
+  #
+  # message - A Message instance. Listeners can flag this message as 'done' to
+  #           prevent further execution.
+  #
+  # cb - Optional callback that is called when message processing is complete
+  #
+  # Returns nothing.
```

Comment:
I see a few other places where we `return undefined` explicitly. I think otherwise, it'd return whatever async.detectServies` returns (maybe that's nothing, hard to tell from the code)